src/player_manager.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+from pyglet.input import ControllerManager
+from pyglet.window.key import KeyStateHandler
+
+from player import Player
+from player_input import PlayerInput, bind_to_keyboard, set_default_controller_layout
+
+PLAYER_COUNT = 4
+KEYBOARD_PLAYER_INDEX = 0
+KEYBOARD_PLAYER_ALSO_USES_A_CONTROLLER = True
+"""
+If false, keyboard player will be controlled by keyboard, *not* a controller.
+Useful if you want to debug two players and you only have one controller.
+"""
+
+
+class PlayerManager:
+    """
+    Responsible for allocating Players when the game starts,
+    and maintaining a list of all players.
+
+    In the future, may also be responsible for facilitating players joining /
+    leaving the match.
+    """
+
+    def __init__(self, keyboard: KeyStateHandler) -> None:
+        self.players: list[Player] = []
+        """
+        Guaranteed to never be replaced, only mutated.
```

Comment:
probably want to use a getter property exposing an internal `self._players` list to better enforce this convention